org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModelOperation.java

```diff
@@ -805,12 +805,13 @@ public void runOperation(IProgressMonitor monitor) throws JavaModelException {
 		}
 		try {
 			if (isReadOnly()) {
-				run(monitor);
+				JavaCore.runReadOnly(() -> run(monitor));
```

Comment:
Thats interesting: the failing tests - like JavaSearchMultipleProjectsTests.testFieldOccurencesInWorkingCopies() - show that isReadOnly() in this context only means no resources are modified - but still the javamodel is modified.
I will remove the runReadOnly() here.

Reply:
hm, that's also a design problem because if `isReadOnly()` can be wrong (e.g. if a new subclass forgot to override it and set it to return `true`) then the verification mechanism in `JavaCore::runReadOnly` would not throw an exception. Isn't there a way to check for the Deltas of the `JavaModelManager` instead of relying on `isReadOnly()`?